packages/wdio-browserstack-service/src/service.ts

```diff
@@ -95,7 +97,12 @@ export default class BrowserstackService implements Services.ServiceInstance {
             }
         } else {
             // Mocha
-            this._fullTitle = `${test.parent} - ${test.title}`
+            // eslint-disable-next-line no-lonely-if
+            if (test.parent && test.parent !== this._suiteTitle) {
+                this._fullTitle = `${this._suiteTitle} - ${test.parent}`
```

Comment:
Feels like this needs to happen recursively when there is more than one parent suite, in other words, I do not think this handles cases like this:
```js
describe('my component', () => {
  describe('scenario A', () => {
    describe('and some other condition is truthy', () => {
      it('example', () => {
        // this now has three parents, does this work?
      });
    });
  });
});
```

Reply:
For this example, it will set the session name to: `my component - and some other condition is truthy`. For comparison, the sauce labs service will set the job name to `my component` and add context of `and some other condition is truthy - example`. 

I can't figure out how to obtain `scenario A` from any of the hooks. Any ideas?

Reply:
Another option is to use one of the following as the session name:
- `this._suiteTitle` (which would be "my component" in the above example)
- `test.parent` ("and some other condition is truthy")

or keep the current title (test.parent - test.title) except set it before the test (so users may change it without being overridden).